### PR TITLE
test: migrate magic link tests to hono

### PR DESCRIPTION
**Pull request by danielgent**
PR #3 of the latest round of wrangler migrations

### CODE REVIEW

File: test/integration/flows/magic-link-flow.spec.ts [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+            authorization: `Bearer ${token}`,
+            "tenant-id": "tenantId",
+          },
+        },
+      );
+      expect(resInitialQuery.status).toBe(404);
+
+      const response = await client.passwordless.start.$post(
+        {
+          // ouch! I had this typed as "body"... no typesafety...
```
Comment (danielgent) [inline test/integration/flows/magic-link-flow.spec.ts:44]: :eyes: would be amazing to get typesafety here _and_ not have to do that params trick

File: test/integration/flows/magic-link-flow.spec.ts [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        exp,
+        iat,
+        sid,
+        sub,
+        //
+        family_name,
+        given_name,
+        nickname,
+        picture,
+        locale,
```
Comment (danielgent) [inline test/integration/flows/magic-link-flow.spec.ts:132]: I'll investigate what I've done to need to do this on another PR. Something with that change where we always fetch the id_token...

File: test/integration/flows/magic-link-flow.spec.ts [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+      const link = magicLink!;
+
+      const authenticatePath = link?.split("https://example.com")[1];
+
+      expect(authenticatePath).toContain("/passwordless/verify_redirect");
+
+      const querySearchParams = new URLSearchParams(
+        authenticatePath.split("?")[1],
+      );
+      const query = Object.fromEntries(querySearchParams.entries());
```
Comment (danielgent) [inline test/integration/flows/magic-link-flow.spec.ts:212]: this is one of the few actual changes. we're not calling an actual URL now of course

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**